db/migrate/20190118204247_move_miq_set_memberships.rb

```diff
@@ -0,0 +1,35 @@
+require 'ancestry'
+
+class MoveMiqSetMemberships < ActiveRecord::Migration[5.0]
+  class Relationship < ActiveRecord::Base
+    has_ancestry
+  end
+
+  class MiqSetMembership < ActiveRecord::Base
+  end
+
+  def up
+    say_with_time("Converting set relationships to miq_set_memberships") do
+      # ancestry holds the id for the relationship record that references the set
+      # the parent record contains the set object
+      # because these are trivial trees, we can look up the parent and get the resource_id
+      parent_miq_set_id = "(select rel2.resource_id from relationships rel2 where rel2.id = rels.ancestry::bigint)"
```

Comment:
Minor, for readability:

```suggestion
      parent_miq_set_id = "(SELECT rel2.resource_id FROM relationships rel2 WHERE rel2.id = rels.ancestry::bigint)"
```